rest/model/project.go

```diff
@@ -373,6 +373,33 @@ type APIWorkstationConfig struct {
 	GitClone      *bool                        `bson:"git_clone" json:"git_clone"`
 }
 
+type APIContainerResources struct {
+	MemoryMB *int `bson:"memory_mb" json:"memory_mb"`
+	CPU      *int `bson:"cpu" json:"cpu"`
+}
+
+func (cr *APIContainerResources) BuildFromService(h interface{}) error {
+	var def model.ContainerResources
+	switch h.(type) {
```

Comment:
Type asserting a second time in each switch case is a little awkward; you can assign a typed value on the switch line with something like this:
```go
switch dbContainerResources := h.(type) {
case model.ContainerResources:
    def = dbContainerResources
case *model.ContainerResources:
    def = *dbContainerResources
default:
    return errors.Errorf("...")
}
```

Alternatively, you could get rid of the `interface{}` input entirely and just pass in a `model.ContainerResources` type. The only reason a lot of the models pass in interface and return interfaces is to satisfy [this interface](https://github.com/evergreen-ci/evergreen/blob/b221ba413febb0c7d9d2010acac8f3d662edc09b/rest/model/model.go#L6-L9). However, that model interface is only used in a few specific situations (e.g. admin settings, event logs, some top-level REST models `APITask` which are used in user notifications) For all other settings models like this one, it's not necessary to satisfy that interface.